### DISCUSSION TITLE
FIXT.1.1:135->PSE, event> (Disconnecting: Received logout request: Lower sequence received than expected without PossDup flag.

### DISCUSSION THREAD

**Discussion by atlasifs88** [quickfix-j/quickfixj#599] [General]
We experienced a network hiccup that interrupted the connectivity to the exchange FIX server. This error was encountered upon restarting our FIX server, and the FIX app could no longer establish a connection. 

What is a necessary setting on our FIX server that will allow the connection to the exchange FIX server to be established?

**Comment by chrjohn** [comment]
Hard to tell without seeing the logs and config. It seems you were logging on with too low a sequence number. Did you maybe reset your sequence numbers?

**Comment by atlasifs88** [comment]
This is the error in the QF/J log.

CHECK MSGTYPOE: quickfix.fixt11.Message$Header@18e34a2f
CHEK APP ID==
INSIDE the TOADMIN
<20230110-01:45:22, FIXT.1.1:135->PSE, outgoing> (8=FIXT.1.1^A9=96^A35=A^A34=5^A49=135^A52=20230110-01:45:22.832^A56=PSE^A98=0^A108=30^A553=13501999^A554=kKGpKz6Fuj^A1137=9^A10=248^A)
<20230110-01:45:22, FIXT.1.1:135->PSE, event> (Initiated logon request)
<20230110-01:45:22, FIXT.1.1:135->PSE, incoming> (8=FIXT.1.1^A9=000138^A35=5^A49=PSE^A56=135^A34=80^A52=20230110-01:45:22.860^A58=Lower sequence received than expected without PossDup flag. Expected/Received = 63/5^A10=101^A)
fromAdmin===194 heartbeat
^[[2m2023-01-10 09:45:22.981^[[0;39m ^[[32m INFO [iqmei-fix-service,,,]^[[0;39m ^[[35m26377^[[0;39m ^[[2m---^[[0;39m ^[[2m[XT.1.1:135->PSE]^[[0;39m ^[[36mc.h.i.fix.service.app.IqmeiApplication  ^[[0;39m ^[[2m:^[[0;39m fromAdmin1: Message=8=FIXT.1.1^A9=138^A35=5^A34=80^A49=PSE^A52=20230110-01:45:22.860^A56=135^A58=Lower sequence received than expected without PossDup flag. Expected/Received = 63/5^A10=213^A, SessionId=FIXT.1.1:135->PSE
<20230110-01:45:22, FIXT.1.1:135->PSE, event> (Received logout request: Lower sequence received than expected without PossDup flag. Expected/Received = 63/5)
CHECK MSGTYPOE: quickfix.fixt11.Message$Header@3ad00872
CHEK APP ID==
<20230110-01:45:22, FIXT.1.1:135->PSE, outgoing> (8=FIXT.1.1^A9=49^A35=5^A34=6^A49=135^A52=20230110-01:45:22.981^A56=PSE^A10=171^A)
<20230110-01:45:22, FIXT.1.1:135->PSE, event> (Sent logout response)
<20230110-01:45:22, FIXT.1.1:135->PSE, event> (Disconnecting: Received logout request: Lower sequence received than expected without PossDup flag. Expected/Received = 63/5)
^[[2m2023-01-10 09:45:22.982^[[0;39m ^[[32m INFO [iqmei-fix-service,,,]^[[0;39m ^[[35m26377^[[0;39m ^[[2m---^[[0;39m ^[[2m[XT.1.1:135->PSE]^[[0;39m ^[[36mc.h.i.fix.service.app.IqmeiApplication  ^[[0;39m ^[[2m:^[[0;39m onLogout1: SessionId=FIXT.1.1:135->PSE
<20230110-01:45:55, FIXT.1.1:135->PSE, event> (Pending connection not established after 4001 ms.)
<20230110-01:46:00, FIXT.1.1:135->PSE, event> (Pending connection not established after 9002 ms.)
<20230110-01:46:05, FIXT.1.1:135->PSE, event> (Pending connection not established after 14003 ms.)
<20230110-01:46:10, FIXT.1.1:135->PSE, event> (Pending connection not established after 19004 ms.)
<20230110-01:46:15, FIXT.1.1:135->PSE, event> (Pending connection not established after 24005 ms.)
<20230110-01:46:20, FIXT.1.1:135->PSE, event> (Pending connection not established after 29005 ms.)
<20230110-01:46:25, FIXT.1.1:135->PSE, event> (Pending connection not established after 34006 ms.)
<20230110-01:46:30, FIXT.1.1:135->PSE, event> (Pending connection not established after 39007 ms.)
<20230110-01:46:35, FIXT.1.1:135->PSE, event> (Pending connection not established after 44008 ms.)
<20230110-01:46:40, FIXT.1.1:135->PSE, event> (Pending connection not established after 49009 ms.)
~

**Comment by atlasifs88** [comment]
This is our setting

ResetOnLogon = N
ResetOnLogout = N
ResetOnDisconnect = N
ResetSeqNumFlag = N

**Comment by chrjohn** [comment]
Could you also provide the log after the last message before the disconnection was sent up to the point that you already posted above?

**Comment by atlasifs88** [comment]
The QF/J failed to connect to the exchange because of a network issue.  The exchange ended up resetting the port we connected to before the connectivity was re-established. 

<20230110-01:00:00, FIXT.1.1:135->PSE, event> (MINA session created: local=/XX.XXX.XX.241:XXXXX, class org.apache.mina.transport.socket.nio.NioSocketSession, remote=/XX.4.XXX.XX:XXXXX)
<20230110-01:00:00, FIXT.1.1:135->PSE, event> (Session state is not current; resetting FIXT.1.1:135->PSE)
CHECK MSGTYPOE: quickfix.fixt11.Message$Header@373f7ceb
CHEK APP ID==
INSIDE the TOADMIN
<20230110-01:00:00, FIXT.1.1:135->PSE, outgoing> (8=FIXT.1.1^A9=96^A35=A^A34=1^A49=135^A52=20230110-01:00:00.511^A56=PSE^A98=0^A108=30^A553=99999999^A554=XXXXXXXX^A1137=9^A10=225^A)
<20230110-01:00:00, FIXT.1.1:135->PSE, event> (Initiated logon request)
<20230110-01:00:02, FIXT.1.1:135->PSE, error> (Disconnecting: Timed out waiting for logon response)
^[[2m2023-01-10 09:00:02.496^[[0;39m ^[[32m INFO [iqmei-fix-service,,,]^[[0;39m ^[[35m22707^[[0;39m ^[[2m---^[[0;39m ^[[2m[      QFJ Timer]^[[0;39m ^[[36mc.h.i.fix.service.app.IqmeiApplication  ^[[0;39m ^[[2m:^[[0;39m onLogout1: SessionId=FIXT.1.1:135->PSE
^[[2m2023-01-10 09:00:30.448^[[0;39m ^[[32m INFO [iqmei-fix-service,,,]^[[0;39m ^[[35m22707^[[0;39m ^[[2m---^[[0;39m ^[[2m[trap-executor-0]^[[0;39m ^[[36mc.n.d.s.r.aws.ConfigClusterResolver     ^[[0;39m ^[[2m:^[[0;39m Resolving eureka endpoints via configuration
<20230110-01:00:34, FIXT.1.1:135->PSE, event> (Pending connection not established after 4002 ms.)
<20230110-01:00:39, FIXT.1.1:135->PSE, event> (Pending connection not established after 9003 ms.)
<20230110-01:00:44, FIXT.1.1:135->PSE, event> (Pending connection not established after 14003 ms.)
<20230110-01:00:49, FIXT.1.1:135->PSE, event> (Pending connection not established after 19004 ms.)
<20230110-01:00:54, FIXT.1.1:135->PSE, event> (Pending connection not established after 24005 ms.)
<20230110-01:00:59, FIXT.1.1:135->PSE, event> (Pending connection not established after 29007 ms.)
<20230110-01:01:04, FIXT.1.1:135->PSE, event> (Pending connection not established after 34008 ms.)

**Reply by atlasifs88** [reply]
Today's log after the FIX server was restarted before market opens.

Notice that there is a resend request.

<20230112-01:10:10, FIXT.1.1:135->PSE, event> (Sent ResendRequest FROM: 1 TO: 63)



20230112-01:10:09, FIXT.1.1:135->PSE, event> (MINA session created: local=/XX.XXX.XX.XX:XXX40, class org.apache.mina.transport.socket.nio.NioSocketSession, remote=/XX.XXX.XX.XX:XXX31)
CHECK MSGTYPOE: quickfix.fixt11.Message$Header@9f292d7
CHEK APP ID==
INSIDE the TOADMIN
<20230112-01:10:10, FIXT.1.1:135->PSE, outgoing> (8=FIXT.1.1^A9=96^A35=A^A34=1^A49=135^A52=20230112-01:10:10.754^A56=PSE^A98=0^A108=30^A553=XXXXX999^A554=XXXXXXXXX^A1137=9^A10=238^A)
<20230112-01:10:10, FIXT.1.1:135->PSE, event> (Initiated logon request)
<20230112-01:10:10, FIXT.1.1:135->PSE, incoming> (8=FIXT.1.1^A9=000069^A35=A^A49=PSE^A56=135^A34=64^A52=20230112-01:10:10.784^A98=0^A108=30^A1137=8^A10=000^A)
<20230112-01:10:10, FIXT.1.1:135->PSE, event> (Setting DefaultApplVerID (1137=8) from Logon)
fromAdmin===194 heartbeat
^[[2m2023-01-12 09:10:10.967^[[0;39m ^[[32m INFO [iqmei-fix-service,,,]^[[0;39m ^[[35m20367^[[0;39m ^[[2m---^[[0;39m ^[[2m[XT.1.1:135->PSE]^[[0;39m ^[[36mc.h.i.fix.service.app.IqmeiApplication  ^[[0;39m ^[[2m:^[[0;39m fromAdmin1: Message=8=FIXT.1.1^A9=69^A35=A^A34=64^A49=PSE^A52=20230112-01:10:10.784^A56=135^A98=0^A108=30^A1137=8^A10=064^A, SessionId=FIXT.1.1:135->PSE

**_**<20230112-01:10:10, FIXT.1.1:135->PSE, event> (Received logon)
<20230112-01:10:10, FIXT.1.1:135->PSE, event> (MsgSeqNum too high, expecting 1 but received 64:**_** 

8=FIXT.1.1^A9=69^A35=A^A34=64^A49=PSE^A52=20230112-01:10:10.784^A56=135^A98=0^A108=30^A1137=8^A10=064^A)
<20230112-01:10:10, FIXT.1.1:135->PSE, event> (Enqueued at pos 64: 8=FIXT.1.1^A9=69^A35=A^A34=64^A49=PSE^A52=20230112-01:10:10.784^A56=135^A98=0^A108=30^A1137=8^A10=064^A)
CHECK MSGTYPOE: quickfix.fixt11.Message$Header@4e1615bf
CHEK APP ID==
<20230112-01:10:10, FIXT.1.1:135->PSE, outgoing> (8=FIXT.1.1^A9=58^A35=2^A34=2^A49=135^A52=20230112-01:10:10.969^A56=PSE^A7=1^A16=0^A10=028^A)
<20230112-01:10:10, FIXT.1.1:135->PSE, event> (Sent ResendRequest FROM: 1 TO: 63)

**Reply by chrjohn** [reply]
Too high a sequence number is no problem since it will be able to sync on its own via resend requests
